clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -715,6 +715,12 @@ public void abortTransaction() throws ProducerFencedException {
      * producer and create a new instance.
      * </p>
      * <p>
+     * If the message format of the destination topic is not upgraded to 0.11.0.0, produce requests will
```

Comment:
Do we need to make it clear that this is only the case if idempotence=true? And do we want to add additional @throws lines?